collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -746,6 +747,29 @@ protected void onBackPressedWithoutLogger() {
         super.onBackPressed();
     }
 
+    private void navigateToTheLastRelevantIndex(FormController formController) {
```

Comment:
I think this is a candidate for moving to JavaRosa's `FormEntryController`. It could become something like `jumpToIndexOrFirstPriorAvailable`. If you agree, please file an issue there. The two advantages are that it reduces complexity in Collect and that we would be able to use really fast unit tests for testing it.

Reply:
That seems fine I'll file an issue.